paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/DynamicBucketSink.java

```diff
@@ -62,17 +62,23 @@ public DataStreamSink<?> build(DataStream<T> input, @Nullable Integer parallelis
         // committer
 
         // 1. shuffle by key hash
-        Integer assignerParallelism = table.coreOptions().dynamicBucketAssignerParallelism();
-        if (assignerParallelism == null) {
-            assignerParallelism = parallelism;
+        Integer numAssigners = table.coreOptions().dynamicBucketAssignerParallelism();
+        if (numAssigners == null) {
+            numAssigners = parallelism;
         }
+
+        Integer totalAssigners = numAssigners;
+        if (parallelism != null && parallelism > totalAssigners) {
+            totalAssigners = parallelism;
+        }
+
         DataStream<T> partitionByKeyHash =
-                partition(input, channelComputer1(), assignerParallelism);
+                partition(input, assignerChannelComputer(numAssigners), totalAssigners);
```

Comment:
This is a little weried for me. From the name, `dynamic-bucket.assigner-parallelism` should control the parallelism of the assigner operator, but it is not anymore. If I want to increase the parallelism of the assigner operator, then I can only increase the parallelism of the sink? 

How about introducing a new configuration to control the initial number of buckets?

Reply:
I agree with you, your point is better.